utils/digital_interrupts.go

```diff
@@ -11,15 +12,46 @@ import (
 	"go.viam.com/rdk/resource"
 )
 
-// DigitalInterruptConfig describes the configuration of digital interrupt for the board.
-type DigitalInterruptConfig struct {
-	Name string `json:"name"`
-	Pin  string `json:"pin"`
-	Type string `json:"type,omitempty"` // e.g. basic, servo
+// PinConfig describes the configuration of a pin for the board
+type PinConfig struct {
+	Name       string  `json:"name"`
+	Pin        string  `json:"pin"`
+	Type       PinType `json:"type,omitempty"`        // e.g. gpio, interrupt
+	DebounceMS int     `json:"debounce_ms,omitempty"` // only used with interrupts
+	PullState  Pull    `json:"pull,omitempty"`
+}
+
+// PinType defines the pin types we support.
+type PinType string
+
+// PinGPIO represents GPIO pins.
+const PinGPIO PinType = "gpio"
+
+// PinInterrupt represents interrupt pins
+const PinInterrupt PinType = "interrupt"
+
+// Pull defines the pins pull state(pull up vs pull down)
+type Pull string
+
+const PullUp Pull = "up"
+const PullDown Pull = "down"
+const PullNone Pull = "none"
+const PullDefault Pull = ""
```

Comment:
```suggestion
// PinType represents the type of pin (GPIO or Interrupt)
type PinType string

const (
    PinGPIO      PinType = "gpio"
    PinInterrupt PinType = "interrupt"
)

// Pull defines the pin's pull state (up, down, or none)
type Pull string

const (
    PullUp      Pull = "up"
    PullDown    Pull = "down"
    PullNone    Pull = "none"
    PullDefault Pull = ""
)
```

This way it's a lot cleaner

Reply:
@JohnN193 shouldn't the default be "none"? why is there an ""?

Reply:
the distinction is choosing to set none, rather than telling the raspberry pi to set whatever is the default/already configured